python/__init__.py

```diff
@@ -0,0 +1,405 @@
+import functools
+import os
+import sysconfig
+import subprocess
+import tempfile
+from pathlib import Path
+
+from triton.common.backend import BaseBackend, register_backend
+from triton.compiler.make_launcher import make_so_cache_key
+from triton.runtime.cache import get_cache_manager
+from triton.runtime.jit import version_key
+
+
+def _get_triton_shared_opt_path() -> str:
+    path = os.getenv("TRITON_SHARED_OPT_PATH", "")
+    if path == "":
+        assert Exception("TRITON_SHARED_OPT_PATH is not set.")
+    return path
+
+
+def _get_llvm_bin_path(bin_name: str) -> str:
+    path = os.getenv("LLVM_BINARY_DIR", "")
+    if path == "":
+        raise Exception("LLVM_BINARY_DIR is not set.")
+    return f"{path}/{bin_name}"
+
+
+def _ttir_to_ttsharedir(mod):
+    # Get Triton-MLIR as string
+    ttir_code = str(mod)
+    with tempfile.TemporaryDirectory() as tmpdir:
+        src_path = os.path.join(tmpdir, "tt.mlir")
+        dst_path = os.path.join(tmpdir, "ttshared.mlir")
+        Path(src_path).write_text(ttir_code)
+        triton_shared_opt_path = _get_triton_shared_opt_path()
+        ret = subprocess.check_call([triton_shared_opt_path, src_path, "--triton-to-linalg", "-o", dst_path])
+        assert ret == 0
```

Comment:
nit: `subprocess.check_call` always returns 0 or raises an exception otherwise, we can remove all of the asserts😄

Reply:
Thanks! I didn't know that. Removed all of the `ret == 0` patterns for `subprocess.check_call()`.